exasol/analytics/udf/communication/broadcast_operation.py

```diff
@@ -54,8 +55,15 @@ def _receive_from_localhost_leader(self) -> bytes:
         self._check_sequence_number(specific_message_obj=specific_message_obj)
         return frames[1].to_bytes()
 
+    @property
```

Comment:
Property method `multi_node_communicator` checks the private attribute `_multi_node_communicator` and raises an exception when it is `None` to avoid code duplication.

Questions: 
* Q1) The names of the private attribute and the property differ on in the prefix `_` and potentially could be mixed up. Should we use different names that can easier be distinguished? 
* Q2) The property is public. Should it be private?

Reply:
See last push